File: src/main/java/com/gitlabci/plugin/language/GitlabYamlFileType.java
package com.gitlabci.plugin.language;

import icons.GitlabCIPluginIcons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class GitlabYamlFileType extends LanguageFileType {

    public static final GitlabYamlFileType INSTANCE = new GitlabYamlFileType();

    private GitlabYamlFileType() {
        super(GitlabYamlLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Gitlab YAML File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Gitlab YAML File";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "gitlab-ci.yml";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return GitlabCIPluginIcons.GitlabYamlIcon;
    }

}

File: src/main/kotlin/com/github/fiskirton/gitlabciplugin/language/GitlabYamlFileType.kt
package com.github.fiskirton.gitlabciplugin.language

import com.intellij.openapi.fileTypes.LanguageFileType
import icons.GitlabCIPluginIcons
import javax.swing.Icon

class GitlabYamlFileType private constructor() : LanguageFileType(GitlabYamlLanguage.INSTANCE) {
    override fun getName(): String {
        return "Gitlab YAML File"
    }

    override fun getDescription(): String {
        return "Gitlab YAML File"
    }

    override fun getDefaultExtension(): String {
        return "gitlab-ci.yml"
    }

    override fun getIcon(): Icon {
        return GitlabCIPluginIcons.GitlabYamlIcon
    }

    companion object {
        val INSTANCE = GitlabYamlFileType()
    }
}